rllib/examples/random_parametric_agent.py

```diff
@@ -64,29 +66,31 @@ def set_weights(self, weights):
         pass
 
 
-# Create a new Trainer using the Policy and config defined above and a new
-# execution plan.
+# Backward compatibility, just in case users want to use the erroneous old name.
+RandomParametriclPolicy = RandomParametricPolicy
+
+
 class RandomParametricTrainer(Trainer):
+    """Trainer with Policy and config defined above and overriding `training_iteration`.
+
+    Overrides the `training_iteration` method, which only runs a (dummy)
+    rollout and performs no learning.
+    """
+
     @classmethod
     def get_default_config(cls):
         return DEFAULT_CONFIG
 
     def get_default_policy_class(self, config):
-        return RandomParametriclPolicy
-
-    @staticmethod
-    def execution_plan(
-        workers: WorkerSet, config: TrainerConfigDict, **kwargs
-    ) -> LocalIterator[dict]:
-        rollouts = ParallelRollouts(workers, mode="async")
+        return RandomParametricPolicy
 
-        # Collect batches for the trainable policies.
-        rollouts = rollouts.for_each(
-            SelectExperiences(local_worker=workers.local_worker())
-        )
+    @override(Trainer)
+    def training_iteration(self) -> ResultDict:
+        # Perform rollouts (only for collecting metrics later).
+        _ = synchronous_parallel_sample(self.workers)
 
-        # Return training metrics.
-        return StandardMetricsReporting(rollouts, workers, config)
+        # Return (empty) training metrics.
+        return {}
```

Comment:
why not collect rollout related metrics here?

Reply:
B/c it's done automatically by RLlib after this. So we always just return the learner stats here.
But yes, we should start thinking about a way to customize this bit of the iteration.